src/jarabe/view/viewsource.py

```diff
@@ -371,6 +385,31 @@ def __init__(self, file_name, document_path, title, bundle=False):
         box.append_item(menu_item)
         menu_item.show()
 
+    def __show_duplicate_alert(self, menu_item):
+        alert = ConfirmationAlert()
+        alert.props.title = _('Duplicate activity')
+        alert.props.msg = _('This may take a few minutes')
+        alert.connect('response', self.__duplicate_alert_response_cb)
+        self.get_toplevel().add_alert(alert)
+
+    def __duplicate_alert_response_cb(self, alert, response_id):
+        self.get_toplevel().remove_alert(alert)
+
+        if response_id == Gtk.ResponseType.OK:
+            self.__set_busy_cursor(True)
+
+            def internal_callback(new_alert):
+                GObject.idle_add(self.__copy_to_home_cb,
+                                 None,
+                                 new_alert)
+
+            new_alert = Alert()
+            new_alert.props.title = _("Duplicate activity")
+            new_alert.props.msg = _("Copying...")
+
+            self.get_toplevel().add_alert(new_alert)
+            GObject.idle_add(internal_callback, new_alert)
```

Comment:
Ah.
That happens when you try to port another patch to here :laughing: 

It was:

+            def internal_callback():
+                self.emit('remove-alert', alert)
+                GObject.idle_add(self.__copy_to_home_cb, new_alert)

It should nt idle_add now. fixing it